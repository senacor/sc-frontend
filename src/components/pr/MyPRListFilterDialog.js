import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Icon from '@material-ui/core/Icon';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap'
  }
});

class MyPRListFilterDialog extends React.Component {
  state = {
    open: false
  };

  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes } = this.props;

    return (
      <div>
        <Button onClick={this.handleClickOpen}>
          <Icon className={classes.leftIcon}>filter_list</Icon>Filter
        </Button>
        <Dialog
          disableBackdropClick
          disableEscapeKeyDown
          open={this.state.open}
          onClose={this.handleClose}
        >
          <DialogTitle>Filtern der PR-Ansicht</DialogTitle>
          <DialogContent>{this.props.children}</DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} color="primary">
              Ok
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

MyPRListFilterDialog.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(MyPRListFilterDialog);
