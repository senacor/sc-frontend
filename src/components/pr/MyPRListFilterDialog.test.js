import React from 'react';
import MyPRListFilterDialog from './MyPRListFilterDialog';
import { createShallow } from '@material-ui/core/test-utils';

describe('MyPRListFilterDialog Component', () => {
  let shallow = createShallow({ dive: true });

  it('displays the dialog for filtering in mobile view', () => {
    const component = shallow(<MyPRListFilterDialog />);

    expect(component).toMatchSnapshot();
  });
});
