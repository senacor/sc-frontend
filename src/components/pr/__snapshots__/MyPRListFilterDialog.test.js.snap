// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPRListFilterDialog Component should match snapshot 1`] = `
<React.Fragment>
  <WithMobileDialog
    width="md"
  />
  <EventListener
    onResize={[Function]}
    target="window"
  />
</React.Fragment>
`;
